### PR TITLE
[16.0][ADD] purchase_report_shipping_address

### PR DESCRIPTION
**Pull request by AungKoKoLin1997**
This module adds warehouse address details with translatable in warehouse to show in purchase report.

In some countries where the dominant language is not based on Roman alphabet,
it is often necessary to be able to print the shipping address depends on supplier's language.
So, warehouse address details will be displayed in purchase report as a shipping address.

@qrtl

### CODE REVIEW

File: purchase_report_shipping_address/reports/purchase_quotation_templates.xml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+                <t t-set="information_block">
+                    <strong>Shipping address:</strong>
+                    <div>
+                        <span
+                            t-field="o.picking_type_id.warehouse_id.warehouse_address_details"
+                        />
+                    </div>
+                </t>
+            </t>
+        </xpath>
```
Comment (yostashiro) [inline purchase_report_shipping_address/reports/purchase_quotation_templates.xml]:
```suggestion
        <xpath expr="//div[@t-if='o.dest_address_id']" position="attributes">
            <attribute name="t-att-style">'display: none!important' if o.picking_type_id.warehouse_id.warehouse_address_details else ''</attribute>
        </xpath>
        <xpath expr="//div[@t-if='o.dest_address_id']" position="after">
            <div t-else="">
                <span
                    t-field="o.picking_type_id.warehouse_id.warehouse_address_details"
                />
            </div>
        </xpath>
```
Can you try and see if this works.  Do the same for purchase order if it does.

File: purchase_report_shipping_address/readme/DESCRIPTION.rst [hunk 2]
```diff
@@ -0,0 +1,5 @@
+This module adds warehouse address details with translatable in warehouse to show in purchase report.
+
+In some countries where the dominant language is not based on Roman alphabet,
+it is often necessary to be able to print the shipping address depends on supplier's language.
+So, warehouse address details will be displayed in purchase report as a shipping address.
```
Comment (yostashiro) [inline purchase_report_shipping_address/readme/DESCRIPTION.rst]:
```suggestion
It is necessary to be able to print the purchase documents in supplier's language,
however Odoo by defaultit does not come with the ability to present the warehouse address
in different languages.  This module intends to take care of this shortcoming.
```

**Review by yostashiro — CHANGES_REQUESTED** [review]
Please add CONFIGURE.rst.

File: purchase_report_shipping_address/readme/CONFIGURE.rst [hunk 3]
```diff
@@ -0,0 +1,3 @@
+#. Go to *Inventory > Configuration > Warehouses*.
+#. Access one of them clicking on it.
+#. Add address in warehouse address details field.
```
Comment (yostashiro) [inline purchase_report_shipping_address/readme/CONFIGURE.rst]:
```suggestion
#. Go to *Inventory > Configuration > Warehouses* and open a warehouse record for which the address presentation should be adjusted for purchase reports.
#. Update Warehouse Address Details field.
```

File: purchase_report_shipping_address/readme/DESCRIPTION.rst [hunk 4]
```diff
@@ -0,0 +1,5 @@
+This module adds warehouse address details with translatable in warehouse to show in purchase report.
```
Comment (yostashiro) [inline purchase_report_shipping_address/readme/DESCRIPTION.rst]:
```suggestion
This module adds the translatable Warehouse Address Details field in warehouse to show in purchase reports.
```

File: purchase_report_shipping_address/readme/DESCRIPTION.rst [hunk 5]
```diff
@@ -0,0 +1,5 @@
...
+
+It is necessary to be able to print the purchase documents in supplier's language,
+however Odoo by defaultit does not come with the ability to present the warehouse address
+in different languages.  This module intends to take care of this shortcoming.
```
Comment (yostashiro) [inline purchase_report_shipping_address/readme/DESCRIPTION.rst]:
```suggestion
It is necessary to be able to print the purchase documents in supplier's language,
however Odoo by defaultit does not come with the ability to present the warehouse address
in different languages.  This module intends to take care of this shortcoming.
```
```suggestion
Purchase documents are printed in supplier's language, however Odoo by default does not
come with the ability to present the warehouse address in different languages.
This module intends to take care of this shortcoming.
```

File: purchase_report_shipping_address/reports/purchase_order_templates.xml [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+                name="t-att-style"
+            >'display: none!important' if o.picking_type_id.warehouse_id.warehouse_address_details else ''</attribute>
+        </xpath>
+        <xpath expr="//div[@t-if='o.dest_address_id']" position="after">
+            <div t-if="o.picking_type_id.warehouse_id.warehouse_address_details">
+                <span
+                    t-field="o.picking_type_id.warehouse_id.warehouse_address_details"
+                />
+            </div>
+        </xpath>
```
Comment (yostashiro) [inline purchase_report_shipping_address/reports/purchase_order_templates.xml]:
```suggestion
```
I think this div (i.e. o.dest_address_id is true) is for dropship case and therefore should be intact.  Let me know if I've misunderstood something.

File: purchase_report_shipping_address/reports/purchase_quotation_templates.xml [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+                name="t-att-style"
+            >'display: none!important' if o.picking_type_id.warehouse_id.warehouse_address_details else ''</attribute>
+        </xpath>
+        <xpath expr="//div[@t-field='o.dest_address_id']" position="after">
+            <div t-if="o.picking_type_id.warehouse_id.warehouse_address_details">
+                <span
+                    t-field="o.picking_type_id.warehouse_id.warehouse_address_details"
+                />
+            </div>
+        </xpath>
```
Comment (yostashiro) [inline purchase_report_shipping_address/reports/purchase_quotation_templates.xml]:
```suggestion
```
Same as above.

**Review by yostashiro — CHANGES_REQUESTED** [review]

**Review by yostashiro — APPROVED** [review]
Tested on runboat. 👍 

@AungKoKoLin1997 Please squash your commits.

**Review by kakurai8 — APPROVED** [review]
It works as expected. LGTM

**Review by rinaldifirdaus — APPROVED** [review]
Functional test: Working as expected

**Pull request merged**